test/Feature/HLSLLib/sign.32.test

```diff
@@ -0,0 +1,181 @@
+#--- source.hlsl
+
+StructuredBuffer<int4> In1 : register(t0);
+StructuredBuffer<uint4> In2 : register(t1);
+StructuredBuffer<float4> In3 : register(t2);
+StructuredBuffer<float> In4 : register(t3);
+RWStructuredBuffer<int4> Out1 : register(u4);
+RWStructuredBuffer<int4> Out2 : register(u5);
+RWStructuredBuffer<int4> Out3 : register(u6);
+RWStructuredBuffer<uint> Out4 : register(u7);
+
+[numthreads(1,1,1)]
+void main() {
+  // int
+  Out1[0] = sign(In1[0]);
+  int4 Tmp = {sign(In1[0].xyz), sign(In1[0].w)};
+  Out1[1] = Tmp;
+  Out1[2].xy = sign(In1[0].xy);
+
+  // uint
+  Out2[0] = sign(In2[0]);
+  int4 Tmp2 = {sign(In2[0].xyz), sign(In2[0].w)};
+  Out2[1] = Tmp2;
+  Out2[2].xy = sign(In2[0].xy);
+
+  // float
+  Out3[0] = sign(In3[0]);
+  int4 Tmp3 = {sign(In3[1].xyz), sign(In3[1].w)};
+  Out3[1] = Tmp3;
+  Out3[2].xy = sign(In3[0].xy);
+
+  int X = sign(In4[0]); // testing nan
+  Out4[0] = X == 1 || X == 0 || X == -1;
```

Comment:
Small nit - enclosing the bool expressions in parentheses makes this easier to read:
```suggestion
  Out4[0] = (X == 1 || X == 0 || X == -1);
```